contracts/Ethleverage.sol

```diff
@@ -0,0 +1,292 @@
+  address public unknownContract = 0xc936749D2D0139174EE0271bD28325074fdBC654; //https://kovan.etherscan.io/address/0xc936749d2d0139174ee0271bd28325074fdbc654
+
+  uint public makerLR;
+  uint public ethCap = 10000;
+  uint public layers = 3;
+
+
+  //Modifiers
+  modifier onlyOwner {
+    require(msg.sender == owner);
+      _;
+  }
+
+  modifier onlyInvestor {
+    require(investors[msg.sender].prinContr != 0);
+    _;
+  }
+
+  //Functions
+  function Ethleverage(
+    address _CDPaddr,
+    address _Daiaddr,
+    address _wethaddr,
+    address _pethaddr,
+    address _mkrContract,
+    uint _liquidationRatio
+  ) public {
+    owner = msg.sender;
+    CDPContract = _CDPaddr;
+    DaiContract = _Daiaddr;
+    wethContract = _wethaddr;
+    pethContract = _pethaddr;
+    MKRContract = _mkrContract;
+    makerLR = _liquidationRatio;
+  }
+
+	function initialize() public {
+      //for some reason the variables ARE initialized in the constructor function, but an error is thrown without reassigning them here.
+      owner = msg.sender;
+      moneyMakerKovan = 0xd87856163f409777df41DDFBF37A66369E028FA9;
+      liquidKovan =  0xc936749D2D0139174EE0271bD28325074fdBC654; //https://github.com/makerdao/sai/blob/4b0c94b8ef2d8e0951dd0a0eee7c0fce5f5dbb49/src/tap.sol
+      unknownContract = 0xc936749D2D0139174EE0271bD28325074fdBC654; //https://kovan.etherscan.io/address/0xc936749d2d0139174ee0271bd28325074fdbc654
+      makerLR = 151;
+      ethCap = 10000;
+      layers = 3;
+      DaiContract = 0xC4375B7De8af5a38a93548eb8453a498222C4fF2;
+      CDPContract = 0xa71937147b55Deb8a530C7229C442Fd3F31b7db2;
+      wethContract = 0xd0A1E359811322d97991E03f863a0C30C2cF029C;
+      pethContract = 0xf4d791139cE033Ad35DB2B2201435fAd668B1b64;
+      MKRContract = 0xAaF64BFCC32d0F15873a02163e7E500671a4ffcD;
+
+    
+			IwethContract(wethContract).approve(CDPContract, ethCap);
+      IwethContract(wethContract).approve(unknownContract, ethCap);
+			IpethContract(pethContract).approve(CDPContract, ethCap);
+      IpethContract(pethContract).approve(unknownContract, ethCap);
+			IDaiContract(DaiContract).approve(CDPContract, ethCap);
+      IDaiContract(DaiContract).approve(unknownContract, ethCap);
+			IMKRContract(MKRContract).approve(CDPContract, ethCap);
+	}
+
+  // for email contract reference: https://github.com/makerdao/sai/blob/master/src/tub.sol
+  // for workflow reference: https://docs.google.com/document/d/1_7pvv49dYJHIlMkaEqBgLOz77F-dt_BImKWRS2ZtFcE/mobilebasic
+
+  /* Workflow
+  1. Convert ETH into WETH,
+  2. Convert WETH into PETH,
+  3. Open CDP
+  4. Deposit PETH into CDP,
+  5. Withdraw DAI,
+  6. Purchase WETH with DAI via decentralized exchange
+  7. Convert WETH into PETH
+  */
+
+  function leverage(uint256 _ethMarketPrice, uint256 _priceFloor) payable public returns (bool sufficient) {
```

Comment:
It's not a priority, but eventually we should get _ethMarketPrice from maker's price oracle.  I think Hernando has worked out how to do that in the price-oracle branch

Reply:
That sounds good to me